### PR TITLE
[GCP] [DNS logs] Don't parse `rdata` when it is empty

### PR DESCRIPTION
**Pull request by zmoog**
## Proposed commit message

Extends the pipeline `script` processor check to process `rdata` field only if it's not null and not an empty string.

A user reported pipeline errors occurring when the `rdata` is an empty string. 

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] ~~I have verified that all data streams collect metrics or logs.~~
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] ~~I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).~~

## How to test this PR locally

```shell
cd packages/gcp
```

```shell
$ elastic-package test pipeline -v  

2023/11/09 13:59:01 DEBUG GET https://127.0.0.1:5601/api/status
--- Test results for package: gcp - START ---
╭─────────┬────────────────────┬───────────┬──────────────────────────┬────────┬──────────────╮
│ PACKAGE │ DATA STREAM        │ TEST TYPE │ TEST NAME                │ RESULT │ TIME ELAPSED │
├─────────┼────────────────────┼───────────┼──────────────────────────┼────────┼──────────────┤
│ gcp     │ audit              │ pipeline  │ test-audit.log           │ PASS   │  15.953834ms │
│ gcp     │ audit              │ pipeline  │ test-sdh-3695.log        │ PASS   │   2.896625ms │
│ gcp     │ compute            │ pipeline  │ test-gcp-compute.json    │ PASS   │      1.758ms │
│ gcp     │ dns                │ pipeline  │ test-dns-empty-rdata.log │ PASS   │   4.668583ms │
│ gcp     │ dns                │ pipeline  │ test-dns.log             │ PASS   │  11.604458ms │
│ gcp     │ firewall           │ pipeline  │ test-firewall.log        │ PASS   │   7.663833ms │
│ gcp     │ loadbalancing_logs │ pipeline  │ test-load-balancer.log   │ PASS   │   3.258583ms │
│ gcp     │ vpcflow            │ pipeline  │ test-vpcflow.log         │ PASS   │  60.242458ms │
╰─────────┴────────────────────┴───────────┴──────────────────────────┴────────┴──────────────╯
--- Test results for package: gcp - END   ---
Done
```

### CODE REVIEW

File: packages/gcp/data_stream/dns/_dev/test/pipeline/test-dns-empty-rdata.log [hunk 1]
```diff
@@ -0,0 +1 @@
+{"insertId":"1f28uoif27xwp0","jsonPayload":{"authAnswer":false,"protocol":"UDP","queryName":"[api.eu](https://api.eu/)-1.crowdstrike.","queryType":"A","rdata":"","responseCode":"NXDOMAIN","serverLatency":2,"sourceNetwork":"security-operations-network"},"logName":"projects/YYY/logs/[dns.googleapis.com](https://dns.googleapis.com/)%2Fdns_queries","receiveTimestamp":"2023-11-09T12:25:20.853671182Z","resource":{"labels":{"location":"europe-west1","project_id":"YYY","source_type":"gce-vm","target_name":"","target_type":"external"},"type":"dns_query"},"severity":"NOTICE","timestamp":"2023-11-09T12:25:19.911990487Z"}
```
Comment (efd6) [inline packages/gcp/data_stream/dns/_dev/test/pipeline/test-dns-empty-rdata.log:1]:
```suggestion
{"insertId":"1f28uoif27xwp0","jsonPayload":{"authAnswer":false,"protocol":"UDP","queryName":"api.eu-1.crowdstrike.","queryType":"A","rdata":"","responseCode":"NXDOMAIN","serverLatency":2,"sourceNetwork":"security-operations-network"},"logName":"projects/YYY/logs/dns.googleapis.com%2Fdns_queries","receiveTimestamp":"2023-11-09T12:25:20.853671182Z","resource":{"labels":{"location":"europe-west1","project_id":"YYY","source_type":"gce-vm","target_name":"","target_type":"external"},"type":"dns_query"},"severity":"NOTICE","timestamp":"2023-11-09T12:25:19.911990487Z"}
```
Just a guess, but it looks like these were copy-paste linkified.

File: packages/gcp/data_stream/dns/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -129,7 +129,7 @@ processors:
       ignore_missing: true
 ### Internal DNS query parsing
   - script:
-      if: ctx?.gcp?.dns?.rdata != null
+      if: ctx?.gcp?.dns?.rdata != null && ctx?.gcp?.dns?.rdata != ""
```
Comment (efd6) [inline packages/gcp/data_stream/dns/elasticsearch/ingest_pipeline/default.yml:132]:
```suggestion
      if: ctx.gcp?.dns?.rdata != null && ctx.gcp.dns.rdata != ""
```

Comment (zmoog) [reply packages/gcp/data_stream/dns/elasticsearch/ingest_pipeline/default.yml]: Thanks! I missed removing the rendundant `?` once again! 🤦  I'll do better next time!

**Review by kaiyan-sheng — APPROVED** [review]
Looks good to me.

**Comment by zmoog** [comment]
Hey @efd6, do you think we should make other changes to this PR, or we're good?

**Review by efd6 — APPROVED** [review]

**Pull request merged**